algorithm/cgal/cgal.cc

```diff
@@ -1705,27 +1705,57 @@ void convertArrangementToPolygonsWithHoles(
     Arrangement_2::Ccb_halfedge_const_circulator start = face->outer_ccb();
     Arrangement_2::Ccb_halfedge_const_circulator edge = start;
     do {
-      if (edge->source()->point() == edge->target()->point()) {
+      const Point_2& point = edge->source()->point();
+      if (point == edge->target()->point()) {
         // Skip zero length edges.
         continue;
       }
-      polygon_boundary.push_back(edge->source()->point());
+      if (polygon_boundary.size() >= 2 &&
+          CGAL::collinear(polygon_boundary.end()[-2],
+                          polygon_boundary.end()[-1], point)) {
+        // Skip colinear points.
+        polygon_boundary.end()[-1] = point;
+      } else {
+        polygon_boundary.push_back(point);
+      }
     } while (++edge != start);
 
+    if (polygon_boundary.size() > 3 &&
+        CGAL::collinear(polygon_boundary.end()[-2], polygon_boundary.end()[-1],
+                        polygon_boundary[0])) {
+      // Skip colinear points.
+      polygon_boundary.resize(polygon_boundary.size() - 1);
+    }
+
     std::vector<Polygon_2> polygon_holes;
     for (Arrangement_2::Hole_const_iterator hole = face->holes_begin();
          hole != face->holes_end(); ++hole) {
       Polygon_2 polygon_hole;
       Arrangement_2::Ccb_halfedge_const_circulator start = *hole;
       Arrangement_2::Ccb_halfedge_const_circulator edge = start;
       do {
-        if (edge->source()->point() == edge->target()->point()) {
+        const Point_2& point = edge->source()->point();
+        if (point == edge->target()->point()) {
           // Skip zero length edges.
           continue;
         }
-        polygon_hole.push_back(edge->source()->point());
+        if (polygon_hole.size() >= 2 &&
+            CGAL::collinear(polygon_hole.end()[-2], polygon_hole.end()[-1],
```

Comment:
This is the part that removed the extra points?

Reply:
Part of it -- it checks sequences of three vertices for colinearity, and replaces or removes the central one if it finds it.